### PR TITLE
Bg 58455 tx to psbt

### PR DESCRIPTION
**Pull request by saravanan7mani**

### CODE REVIEW

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  function getInputScripts(): InputType[][] {
+    return outputScripts.scriptTypes2Of3.flatMap((t) => {
+      return outputScripts.scriptTypes2Of3.flatMap((lastType) => {
+        return [[t, t, lastType]];
+      });
+    });
+  }
+
+  function getSignerPairs(): [signer: KeyName, cosigner: KeyName][] {
+    return [['user', 'bitgo']];
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/test/bitgo/psbt/Psbt.ts:213]:
```suggestion
  function getSignerPairs(containsTaprootInput: boolean): [signer: KeyName, cosigner: KeyName][] {
    return [['user', 'bitgo'], ...containsTaprootInput ? [['user', 'backup']] : []];
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/utxo-lib/src/bitgo/Unspent.ts [hunk 2]
```diff
@@ -138,6 +138,21 @@ export function toPrevOutput<TNumber extends number | bigint>(
...
 
+/**
+ * @return PrevOutput with prevTx from Unspent
+ */
+export function toPrevOutputWithPrevTx<TNumber extends number | bigint>(
+  u: Unspent<TNumber> & { prevTx?: Buffer },
+  network: Network
+): PrevOutput<TNumber> {
+  const prevTx = u.prevTx;
+  return {
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/Unspent.ts]: Done

File: modules/utxo-lib/src/bitgo/wallet/Psbt.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+/**
+ * @return true iff the PSBT input at inputIndex is not signed.
+ */
+export function isPsbtInputUnsigned(psbt: UtxoPsbt<UtxoTransaction<bigint>>, inputIndex: number): boolean {
+  const input = checkForInput(psbt.data.inputs, inputIndex);
+  return (
+    (!Array.isArray(input.partialSig) || input.partialSig.length === 0) &&
+    (!Array.isArray(input.tapScriptSig) || input.tapScriptSig.length === 0)
+  );
+}
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/src/bitgo/wallet/Psbt.ts]: Done

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+    unspents.forEach((u, i) => {
+      if (isWalletUnspent(u)) {
+        try {
+          if (txSignatureStatus === 'unsigned') {
+            signWalletPsbt(psbt, i, rootWalletKeys[signer], u);
+          }
+          signWalletPsbt(psbt, i, rootWalletKeys[cosigner], u);
+        } catch (err) {
+          assert.deepStrictEqual(txSignatureStatus, 'unsigned');
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/test/bitgo/psbt/Psbt.ts]: Done

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        if (isWalletUnspent(u)) {
+          return { ...u, value: BigInt(u.value) };
+        } else {
+          throw new Error(`invalid unspent`);
+        }
+      });
+
+      const psbt = constructPsbtUsingTransaction(txBigInt, unspentsBigInt, signer, cosigner, signatureStatus);
+
+      const contaionsP2trInput = inputScriptTypes.includes('p2tr');
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/test/bitgo/psbt/Psbt.ts]: Done

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  }
+
+  function runTestSignUnspents<TNumber extends number | bigint>({
+    inputScriptTypes,
+    outputScriptType,
+    signer,
+    cosigner,
+    amountType,
+    testOutputAmount,
+    signatureStatus,
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/test/bitgo/psbt/Psbt.ts]: Done

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+        } else {
+          throw new Error(`invalid unspent`);
+        }
+      });
+
+      const psbt = constructPsbtUsingTransaction(txBigInt, unspentsBigInt, signer, cosigner, signatureStatus);
+
+      const contaionsP2trInput = inputScriptTypes.includes('p2tr');
+      if (signatureStatus === 'unsigned' && contaionsP2trInput) {
+        assert.throws(
```
[the remark this answers is not in the document]
Comment (saravanan7mani) [reply modules/utxo-lib/test/bitgo/psbt/Psbt.ts:236]: Done

File: modules/utxo-lib/test/bitgo/psbt/Psbt.ts (hunk 1 above)
Comment (saravanan7mani) [reply modules/utxo-lib/test/bitgo/psbt/Psbt.ts]: Done

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**